brick/hooks/pre_gen.dart

```diff
@@ -1,5 +1,21 @@
 import 'package:mason/mason.dart';
+import 'package:recase/recase.dart';
 
 void run(HookContext context) {
   context.vars['current_year'] = DateTime.now().year;
+
+  context.vars['application_id_android'] = _appId(context, isAndroid: true);
+  context.vars['application_id_ios'] = _appId(context, isAndroid: false);
+}
+
+String _appId(HookContext context, { required bool isAndroid }) {
```

Comment:
To make this more extensible thoughts on passing an `enum` instead of `bool` for `platform`:

```dart
String _appId(HookContext context, {required Platform platform}) {...}
```

It just feels a bit weird to have `isAndroid` imo. Lmk what you think, thanks!

Reply:
Sounds good, honestly I think the bool works great because we just need to know if it is ios or android here, as far as I understand the bundle/application id only exists on those two platforms, so having an enum seemed a little bit of too much since we need only a binary value.

But clearly I am in the minor part here, since Renan also suggested something similar haha, so I am gonna implement it.

Reply:
I think the confusing part is this bool doesn't clearly indicator the it's either iOS or Android. Based on the bool all we know is whether the platform is android or not so the fact that when isAndroid is false the platform is iOS is just an assumption that we make which isn't explicit. Also this wouldn't scale well if we needed to support additional platforms in the future so I feel it'd be a bit more clear to just use an enum to represent the platform here.

Reply:
Alright, you convinced me!